bundler/spec/bundler/source/git/git_proxy_spec.rb

```diff
@@ -3,127 +3,177 @@
 RSpec.describe Bundler::Source::Git::GitProxy do
   let(:path) { Pathname("path") }
   let(:uri) { "https://github.com/rubygems/rubygems.git" }
-  let(:ref) { "HEAD" }
+  let(:ref) { nil }
+  let(:branch) { nil }
+  let(:tag) { nil }
+  let(:options) { { "ref" => ref, "branch" => branch, "tag" => tag }.compact }
   let(:revision) { nil }
   let(:git_source) { nil }
   let(:clone_result) { double(Process::Status, :success? => true) }
   let(:base_clone_args) { ["clone", "--bare", "--no-hardlinks", "--quiet", "--no-tags", "--depth", "1", "--single-branch"] }
-  subject { described_class.new(path, uri, ref, revision, git_source) }
+  subject(:instance) { described_class.new(path, uri, options, revision, git_source) }
```

Comment:
is it necessary to change the name of the `subject` here?  it makes the diff harder to read

Reply:
I am going to change `instance` to `git_proxy`, as that is a more clear name.  Using subject directly is "bad rspec practice" (according to rubocop-rspec), particularly when overriding the subject inside a test , as had been done on [old line 41](https://github.com/rubygems/rubygems/pull/6959/files#diff-9ede4bd79279fea44d1c52a45f234c711d280867436ec01be6c948a17d5a3c93L41).

Reply:
Done, it is now `git_proxy`.